### PR TITLE
[Elastic Log Plugin] Turn on Jenkins and Travis CI tests

### PR DESCRIPTION
**Pull request by fearful-symmetry**
This PR just enables our current unit tests to run under CI, which is probably one of the last things we'll need before we merge this to master.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -325,6 +325,25 @@ pipeline {
             }
           }
         }
+        stage('dockerlogbeat'){
+          agent { label 'ubuntu && immutable' }
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            expression {
+              return env.BUILD_DOCKERLOGBEAT != "false"
```
Comment (exekias) [inline Jenkinsfile]: You need to define this `BUILD_DOCKERLOGBEAT` in the top

**Comment by exekias** [comment]
@kuisathaverat maybe you can help here? I think you know Jenkinsfile very well :innocent:

**Comment by exekias** [comment]
```
Installing mage v1.8.0 from vendor dir.
00:16:51.119  /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-14947/.magefile cleaned
00:16:51.119  mage unitTest
00:17:03.356  2019/12/05 16:25:34 Found Elastic Beats dir at /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-14947/src/github.com/elastic/beats
00:17:03.356  >> go test: Unit Testing
00:17:21.467  SUMMARY:
00:17:21.468    Fail:     0
00:17:21.468    Skip:     0
00:17:21.468    Pass:     2
00:17:21.468    Packages: 2
00:17:21.468    Duration: 19.055671787s
00:17:21.468    JUnit Report: /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-14947/src/github.com/elastic/beats/x-pack/dockerlogbeat/build/TEST-go-unit.xml
00:17:21.468    Output File:  /var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-14947/src/github.com/elastic/beats/x-pack/dockerlogbeat/build/TEST-go-unit.out
00:17:21.468  >> go test: Unit Test Passed
00:17:21.728  ?   	github.com/elastic/beats/x-pack/dockerlogbeat	[no test files]
00:17:21.728  >> python test: Unit Testing
00:17:25.929  DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
00:17:35.913  >> python test: Unit Testing - No tests found.
00:17:35.913  make: Leaving directory '/var/lib/jenkins/workspace/Beats_beats-beats-mbp_PR-14947/src/github.com/elastic/beats/x-pack/dockerlogbeat'
```

It would seem like tests are not passing, right?

**Comment by fearful-symmetry** [comment]
> It would seem like tests are not passing, right?

I'm not sure what you mean?

` Pass:     2`

**Comment by exekias** [comment]
arg true, sorry I was missled by:
```
00:17:21.728  ?   	github.com/elastic/beats/x-pack/dockerlogbeat	[no test files]
```

Where there is no actual line for tests that are passing, :thinking: Perhaps you can push a commit breaking a test to check things work as expected?

**Comment by exekias** [comment]
:clap: LGTM, maybe it's time to go out of draft :P

File: .travis.yml [hunk 2]
```diff
@@ -134,6 +134,12 @@ jobs:
       go: $TRAVIS_GO_VERSION
       stage: test
 
+    # Docker Log Driver
+    - os: linux
+      env: TARGETS="-C x-pack/dockerlogbeat unit-tests"
```
Comment (jsoriano) [inline .travis.yml]: Should we have a `testsuite` target for consistency with other beats? (in this case it will only depend on `unit-tests` target)

Comment (fearful-symmetry) [reply .travis.yml]: I tired to do that, but `testsuite` is a common target imported from the libbeat mage file, and it doesn't seem to work with the docker plugin's wierd design.

Comment (fearful-symmetry) [reply .travis.yml]: Right now we just have unit tests, and the testsuite also wants integration tests and an `update` command.

**Comment by kuisathaverat** [comment]
😢 I guess I've seen it too late because it works fine now, it is nice to see that you could add a new stage without help 😀

**Comment by fearful-symmetry** [comment]
@kuisathaverat everything look good here?

**Pull request merged**